source/common/conn_pool/conn_pool_base.cc

```diff
@@ -559,5 +564,14 @@ void ActiveClient::onConnectTimeout() {
   close();
 }
 
+void ActiveClient::onLifetimeTimeout() {
+  if (state_ != ActiveClient::State::CLOSED) {
```

Comment:
Please make sure capacity stats are handed correctly. This should decrease the local capacity and possibly trigger prefetching.

Reply:
Good call, I'll look into this.